pilot/pkg/xds/monitoring.go

```diff
@@ -65,6 +65,21 @@ var (
 		monitoring.WithLabels(typeTag),
 	)
 
+	// Number of delayed pushes. Currently this happens only when the last push has not been ACKed
+	totalDelayedPushes = monitoring.NewSum(
+		"pilot_xds_delayed_pushes_total",
+		"Total number of XDS pushes that are delayed.",
+		monitoring.WithLabels(typeTag),
+	)
+
+	// Number of delayed pushes that we pushed prematurely as a failsafe.
+	// This indicates that either the failsafe timeout is too aggressive or there is a deadlock
+	totalDelayedPushTimeouts = monitoring.NewSum(
+		"pilot_xds_delayed_push_timeouts_total",
+		"Total number of XDS pushes that are delayed and timed out",
+		monitoring.WithLabels(typeTag),
+	)
+
 	xdsExpiredNonce = monitoring.NewSum(
```

Comment:
Is a gauge for the number proxies falling behind be useful ?

Reply:
I think if we do this we would likely want both, otherwise we cannot tell if the guage is staying at 1 but a bunch of proxies are switching rapidly, for example. I don't mind adding another one although if it would be useful